src/kernels/helpers.ts

```diff
@@ -176,19 +176,20 @@ export function rankKernels(
     }
 
     traceInfoIfCI(`preferredInterpreterKernelSpecIndex = ${preferredInterpreterKernelSpec?.id}`);
+    // traceInfoIfCI(`filtering kernels: ${JSON.stringify(kernels)}`); very large log, but useful for debugging
 
     // Figure out our possible language from the metadata
     const actualNbMetadataLanguage: string | undefined =
         notebookMetadata?.language_info?.name.toLowerCase() ||
         (notebookMetadata?.kernelspec as undefined | IJupyterKernelSpec)?.language?.toLowerCase();
     let possibleNbMetadataLanguage = actualNbMetadataLanguage;
 
+    traceInfoIfCI(`filtering out kernels that don't match notebook's language: ${possibleNbMetadataLanguage}`);
     // If the notebook has a language set, remove anything not that language as we don't want to rank those items
     kernels = kernels.filter((kernel) => {
         if (
             possibleNbMetadataLanguage &&
             possibleNbMetadataLanguage !== PYTHON_LANGUAGE &&
-            !notebookMetadata?.kernelspec &&
```

Comment:
the result of this condition is - if the notebook metadata has a kernelspec, we don't filter based on language